src/test/java/com/google/sps/NaturalLanguagePostprocessorTest.java

```diff
@@ -0,0 +1,165 @@
+        expected.put(ENTITY_A, Arrays.asList(new Long[] {1L}));
+        expected.put(ENTITY_B, Arrays.asList(new Long[] {2L}));
+        expected.put(ENTITY_C, Arrays.asList(new Long[] {3L}));
+
+        Map<String, List<Long>> actual = nlp.getEntitiesMap();
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void SameEntitiesInDifferentTimes() {
+        // Same entities will be found in different times
+
+        // Time 1: [A]
+        // Time 2: [B]
+        // Time 3: [A]
+        // Time 4: [B]
+
+        // Expected: {A: [1, 3], B: [2, 4]}
+        
+        NaturalLanguagePostprocessor nlp = new NaturalLanguagePostprocessor();
+        List<String> time1 = Arrays.asList(new String[] {ENTITY_A}); 
+        List<String> time2 = Arrays.asList(new String[] {ENTITY_B}); 
+        List<String> time3 = Arrays.asList(new String[] {ENTITY_A}); 
+        List<String> time4 = Arrays.asList(new String[] {ENTITY_B}); 
+
+        nlp.addEntities(time1, 1);
+        nlp.addEntities(time2, 2);
+        nlp.addEntities(time3, 3);
+        nlp.addEntities(time4, 4);
+
+        Map<String, List<Long>> expected = new HashMap<>();
+        expected.put(ENTITY_A, Arrays.asList(new Long[] {1L, 3L}));
+        expected.put(ENTITY_B, Arrays.asList(new Long[] {2L, 4L}));
+
+        Map<String, List<Long>> actual = nlp.getEntitiesMap();
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void DifferentEntitiesInSameTimes() {
+        // Different entities will be found in the same times
+
+        // Time 1: [A, B, C]
+        // Time 2: [A, C]
+        // Time 3: [B, C]
+        // Time 4: [A, B]
+
+        // Expected: {A: [1, 2, 4], B: [1, 3, 4], C: [1, 2, 3]}
+        
+        NaturalLanguagePostprocessor nlp = new NaturalLanguagePostprocessor();
+        List<String> time1 = Arrays.asList(new String[] {ENTITY_A, ENTITY_B, ENTITY_C}); 
+        List<String> time2 = Arrays.asList(new String[] {ENTITY_A, ENTITY_C}); 
+        List<String> time3 = Arrays.asList(new String[] {ENTITY_B, ENTITY_C}); 
+        List<String> time4 = Arrays.asList(new String[] {ENTITY_A, ENTITY_B}); 
+
+        nlp.addEntities(time1, 1);
+        nlp.addEntities(time2, 2);
+        nlp.addEntities(time3, 3);
+        nlp.addEntities(time4, 4);
+
+        Map<String, List<Long>> expected = new HashMap<>();
+        expected.put(ENTITY_A, Arrays.asList(new Long[] {1L, 2L, 4L}));
+        expected.put(ENTITY_B, Arrays.asList(new Long[] {1L, 3L, 4L}));
+        expected.put(ENTITY_C, Arrays.asList(new Long[] {1L, 2L, 3L}));
+
+        Map<String, List<Long>> actual = nlp.getEntitiesMap();
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void SameEntitiesInSameTimes() {
+        // The same entities will be found in the same times
+
+        // Time 1: [A, A, A]
+        // Time 2: [B, B]
+        // Time 3: [C]
+        
+        // Expected: {A: [1, 1, 1], B: [2, 2], C: [3]}
```

Comment:
Duplicates in a specified time range are removed in the NaturalLanguageProcessor object, which happens before the postprocessor is called. So when running the input on the entire back-end pipeline, then the expected output is `{A: [1], B: [2], C: [3]}`. However, since this is just a unit test focusing on the prostprocessor and the postprocessor does not concern itself with duplicates, the expected output is `{A: [1, 1, 1], B: [2, 2], C: [3]}`. I mainly included this test to emphasize the postprocessor's functionality when isolated from the rest of the pipeline.